File: src/main/kotlin/de/check24/learningKotlinJava/java/Intersection_of_A_and_B.java
package de.check24.learningKotlinJava.java;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Intersection_of_A_and_B {
    public static void main(String args[]) {
        List A = new ArrayList(Arrays.asList("a","b","c","d","e"));
        List B = new ArrayList(Arrays.asList("f","e","h","b","e"));

        System.out.println("Liste A: " + A);
        System.out.println("Liste B: " + B);

        A.retainAll(B);
        System.out.println("Schnittmenge von A und B: " + A);
    }
}

File: src/main/kotlin/de/check24/learningKotlinJava/kotlin/Intersection_of_A_and_B.kt
package de.check24.learningKotlinJava.kotlin

fun main() {
    val a = mutableListOf("a", "b", "c", "d", "e")
    val b = mutableListOf("f", "e", "h", "b", "e")

    println("Liste A: ${a.intersect(b)}")
    println("Liste B: $b")

    a.retainAll(b) // Retains only the elements in a that are contained in b.
    println("Intersection of a and b: $a")
}
// Ausgabe:
// Liste A: [a, b, c, d, e]
// Liste B: [f, e, h, b, e]
// Intersection of a and b: [b, e]